### ISSUE TITLE
Build failure with 3.x.x

### ISSUE THREAD

**Issue by TacoTheDank** [KeepSafe/dexcount-gradle-plugin#409]
When using any of the 3.x.x releases, using AGP 4.2.x and 7.0.x returns build failures with
```
A problem occurred evaluating project ':app'.
> No signature of method: build_b4h90iu9nq8gldxg84qh9m5wb.android() is applicable for argument types: (build_b4h90iu9nq8gldxg84qh9m5wb$_run_closure1) values: [build_b4h90iu9nq8gldxg84qh9m5wb$_run_closure1@2152a859]

* Exception is:
org.gradle.api.GradleScriptException: A problem occurred evaluating project ':app'.
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:93)
	at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.lambda$apply$0(DefaultScriptPluginFactory.java:133)
	at org.gradle.configuration.ProjectScriptTarget.addConfiguration(ProjectScriptTarget.java:79)
	... (truncated for brevity)

```

Only changed the version number. I'm not sure if there's something I'm missing that's causing this. Latest working version for me  is 2.1.0-RC01.

The project this is happening on is [AntennaPod](https://github.com/AntennaPod/AntennaPod).

**Comment by benjamin-bader** [comment]
Please share your build.gradle demonstrating the error?

**Comment by benjamin-bader** [comment]
Ah, I see the issue.  3.0.0 is an API-breaking release, and in it the `DexCountExtension` class has changed its interface.  Specifically this line: https://github.com/AntennaPod/AntennaPod/blob/develop/app/build.gradle#L79

```groovy
dexcount {
  // Instead of this,
  runOnEachPackage true

  // Do this
  runOnEachPackage = true
}
```

tl;dr the extension now uses Gradle properties exclusively instead of the mixture of methods and kotlin properties from prior releases.

**Comment by TacoTheDank** [comment]
@benjamin-bader I see. Despite reading that it changed from Kotlin to Java, it didn't really occur to me how exactly that would have affected how properties are perceived by Gradle (even though it obviously would have lol). This would be good to note in the 3.0.0 tag release. I think including the example you provided would help other people who might potentially get confused like I did. Thanks!

**Comment by benjamin-bader** [comment]
I thought I did?  The first line in the changelog calls out that this API has changed.

**Comment by benjamin-bader** [comment]
Oh you're talking about the github release itself.  Fair enough; fixed.
